client/state/sites/selectors.js

```diff
@@ -158,6 +188,26 @@ export function getSiteSlug( state, siteId ) {
 }
 
 /**
+ * Returns the domain for a site, or null if the site is unknown.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @return {?String}        Site domain
+ */
+export function getSiteDomain( state, siteId ) {
+	const site = getRawSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+
+	if ( ( site.options && site.options.is_redirect ) || isSiteConflicting( state, siteId ) ) {
```

Comment:
You could reuse `getSiteOption` here:

```js
if ( getSiteOption( state, siteId, 'is_redirect' ) || isSiteConflicting( state, siteId ) ) {
```

At which point you might reorder the function since you're not touching `site` unless these conditions don't apply.

```js
export function getSiteDomain( state, siteId ) {
	if ( getSiteOption( state, siteId, 'is_redirect' ) || isSiteConflicting( state, siteId ) ) {
		return getSiteSlug( state, siteId );
	}
	
	const site = getRawSite( state, siteId );
	if ( ! site ) {
		return null;
	}

	return site.URL.replace( /^https?:\/\//, '' );
}
```